### PR TITLE
Smart contract service review

### LINKED ISSUE #5836: Review hedera-smart-contract-service for problems following merge from develop

### PR DESCRIPTION
**Pull request by mhess-swl**
Closes #5836

### CODE REVIEW

**Review by kimbor — APPROVED** [review]
lgtm!

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public void preHandle(@NonNull final PreHandleContext context) {
         requireNonNull(context);
         final var op = context.getTxn().contractCreateInstanceOrThrow();
         final var adminKey = asHederaKey(op.adminKeyOrElse(Key.DEFAULT));
-        if (adminKey.isPresent() && !((JKey) adminKey.get()).hasContractID()) {
+        if (op.hasAdminKey() && adminKey.isPresent() && !((JKey) adminKey.get()).hasContractID()) {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java:61]: shouldn't `op.hasAdminKey` be present before line 60 ?

Comment (mhess-swl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractCreateHandler.java]: I was thinking it'd be awkward to check, but I found a way that's not so bad. Thanks :)

**Review by povolev15 — APPROVED** [review]
LGTM

**Pull request merged**